backend/app/services/learner_activity_report.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+class LearnerActivityReport
+  def initialize(months_ago:)
+    @months_ago = months_ago
+    @user_uuids = []
+    @csv_string = CSV.generate do |csv|
+      generate_report(csv)
+    end
+  end
+
+  def as_csv_string
+    @csv_string
```

Comment:
this just occurred to me but it might be better to move the block from initialize to here so it's only called when a CSV string is requested.
```
def as_csv_string
    CSV.generate do |csv|
       generate_report(csv)
     end
end
```